pkg/cmd/backuprestore/backupnoconfig.go

```diff
@@ -0,0 +1,246 @@
+package backuprestore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"slices"
+	"strings"
+
+	backupv1alpha1 "github.com/openshift/api/config/v1alpha1"
+	backupv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1alpha1"
+	prune "github.com/openshift/cluster-etcd-operator/pkg/cmd/prune-backups"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+
+	gcron "github.com/go-co-op/gocron/v2"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type backupNoConfig struct {
+	kubeConfig    string
+	snapshotExist bool
+	schedule      string
+	retention     backupv1alpha1.RetentionPolicy
+	backupOptions
+}
+
+func NewBackupNoConfigCommand(errOut io.Writer) *cobra.Command {
+	backupNoConf := &backupNoConfig{
+		snapshotExist: false,
+		backupOptions: backupOptions{errOut: errOut},
+	}
+	cmd := &cobra.Command{
+		Use:   "cluster-backup-no-config",
+		Short: "Backs up a snapshot of etcd database and static pod resources without config",
+		Run: func(cmd *cobra.Command, args []string) {
+			must := func(fn func() error) {
+				if err := fn(); err != nil {
+					if cmd.HasParent() {
+						klog.Fatal(err)
+					}
+					fmt.Fprint(backupNoConf.errOut, err.Error())
+				}
+			}
+
+			must(backupNoConf.Validate)
+			must(backupNoConf.Run)
+		},
+	}
+	backupNoConf.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (b *backupNoConfig) AddFlags(fs *pflag.FlagSet) {
+	b.backupOptions.AddFlags(fs)
+}
+
+func (b *backupNoConfig) Validate() error {
+	return b.Validate()
+}
+
+func (b *backupNoConfig) Run() error {
+	backupsClient, err := b.getBackupClient()
+	if err != nil {
+		return err
+	}
+
+	if err = b.extractBackupSpecs(backupsClient); err != nil {
+		return err
+	}
+
+	var errs []error
+	go func() {
+		err := b.scheduleBackup()
```

Comment:
the cronjob.Start() needs to block this command, otherwise you will have an infinitely restarting sidecar that copies backups all the time

Reply:
the call is non blocking, and there is no return from the `task()` to know when to unblock 

maybe i find another package better

Reply:
so i changed the approach 

- one scheduler being created upon `Run()`
- the scheduler runs only one job `backup()`
- the `prune()` is being invoked upon successful completion of the Job (i.e. backup())
- the scheduler configs is to :
   - run only one job at a time
   - the job run is singleton
   - the job is never being re-queued